packages/core/src/Modal.js

```diff
@@ -0,0 +1,119 @@
+import React, {
+  cloneElement,
+  isValidElement
+} from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import Overlay from './Overlay';
+import HeaderRow from './HeaderRow';
+import TextLabel from './TextLabel';
+import closable from './mixins/closable';
+import icBEM from './utils/icBEM';
+import prefixClass from './utils/prefixClass';
+import renderToLayer from './mixins/renderToLayer';
+
+import './styles/Modal.scss';
+
+export const MODAL_SIZE = ['small', 'large', 'full'];
+
+export const COMPONENT_NAME = prefixClass('modal');
+const ROOT_BEM = icBEM(COMPONENT_NAME);
+export const BEM = {
+    root: ROOT_BEM.modifier('active'),
+    closable: ROOT_BEM.element('closable'),
+    container: ROOT_BEM.element('container'),
+    header: ROOT_BEM.element('header'),
+    body: ROOT_BEM.element('body')
+};
+
+/**
+ * Render Modal Header
+ * If string, render <HeaderRow> with label
+ * If element, render element with headerClassName
+ *
+ * @param  {String|Node|Any} header           - Title string or <HeaderRow>
+ * @param  {String}          headerClassName  - Header className
+ * @return {Node|Any}                         - Header Node or any
+ */
+function renderHeader(header, headerClassName) {
+    if (isValidElement(header)) {
+        return cloneElement(header, {
+            className: headerClassName
+        });
+    } else if (typeof header === 'string') {
+        const label = <TextLabel align="center" basic={header} />;
+        return <HeaderRow className={headerClassName} center={label} />;
+    }
+
+    return header;
+}
+
+export const ModalContent = ({
+    header,
+    bodyClassName,
+    bodyPadding,
+    // React props
+    children,
+}) => (
+    <div className={BEM.container}>
+        {renderHeader(header, `${BEM.header}`)}
+        <div
+            className={classNames(
+                  bodyClassName,
+                  `${BEM.body.modifier('padding', bodyPadding)}`
+              )}>
+            {children}
+        </div>
+    </div>
+    );
+
+ModalContent.propTypes = {
+    header: PropTypes.node,
+    bodyClassName: PropTypes.string,
+    bodyPadding: PropTypes.bool,
+};
+
+ModalContent.defaultProps = {
+    bodyPadding: false,
+};
+
+const ClosableModalContent = closable({
+    onEscape: true,
+    onClickOutside: true,
+    onClickInside: false,
+})(ModalContent);
+
+
+function Modal({
+    size,
+    header,
+    bodyClassName,
+    bodyPadding,
+    // React props
+    className,
+}) {
+    const bemClass = BEM.root.modifier(size);
+    const rootClassName = classNames(bemClass.toString(), className);
+
+    return (
+        <article className={rootClassName}>
+            <Overlay />
+            <ClosableModalContent
```

Comment:
I suggest to forward `closable` prop to `<ClosableModalContent>` here from `<Modal>`, so you can easily override `closable()` configs.

Reply:
Thanks so much! I've added the closable prop back.